File: check_palindrome.js
function checkPalindrome(word){
// function to check if a word is a palindrome or not
    if(word.length==0 || word.length==1){ 
        return true; // true if there is now word or it is of length 1
    }
    
    //compare first character to last character
    if(word.charAt(0) == word.charAt(word.length-1)){
        //slice word
        return checkPalindrome(word.substring(1, word.length-1));
    }
    
    //for false palindromes
    return false;
    
}

//test
console.log(checkPalindrome("kayak")); //prints true
console.log(checkPalindrome("kayam")); //prints false

File: check_palindrome.py
def palindrome_check(word):
    """ Check if word is a palindrome or not. """
    
    if len(word) == 0 or len(word) == 1:
        return True
    elif word[0] == word[len(word)-1]:
        return palindrome_check(word[1:len(word)-1])
    else:    
        return False
    
def main():
    print(palindrome_check("kayak")) # prints true
    
if __name__ == '__main__':
    main()
